tools/update_quest_data.py

```
@@ -0,0 +1,439 @@
+    111: 12,  # MoonCancer
+    112: 13,  # Foreigner
+    113: 14,  # Pretender
+}
+
+
+def fetch_json(url: str, timeout: int = 60, retries: int = 3) -> Optional[dict | list]:
+    """下载 JSON，带重试"""
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    req = urllib.request.Request(url, headers={"User-Agent": "MaaFgo/1.0"})
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)